### PR TITLE
Allow users to define what extensions CA certs will have

### PR DESCRIPTION
**Pull request by ridv**
* New option added to allow users to explicitly set what extensions their CA certs can have.
* Only load files matching those extensions into the Cert pool.

### CODE REVIEW

File: util.go [hunk 1]
```diff
@@ -64,6 +68,42 @@ func init() {
 		AwaitingPulseJobUpdateStates[status] = true
 	}
 }
+func createCertPool(path string, extensions map[string]struct{}) (*x509.CertPool, error) {
+	certPool := x509.NewCertPool()
+
+	_, err := os.Stat(path)
+	if err != nil {
+		return nil, errors.New("given certs path doesn't exist")
```
Comment (JiaminZhu) [inline util.go]: why not include the `err` returned
```
errors.Wrap(err,"given certs path doesn't exist")
```

Comment (misho-kr) [reply util.go]: The error message will be more informative if it contains the path. Mia's suggestion may resolve that.

File: util.go [hunk 2]
```diff
@@ -64,6 +68,42 @@ func init() {
...
+		// Skip any files that do not contain the right extension
+		if _, ok := extensions[filepath.Ext(cert.Name())]; !ok {
+			continue
+		}
+
+		caCert, err := ioutil.ReadFile(filepath.Join(path, cert.Name()))
+		if err != nil {
+			return nil, err
+		}
+		certPool.AppendCertsFromPEM(caCert)
```
Comment (misho-kr) [inline util.go]: This function returns a boolean to indicate if the operation was successful -- https://golang.org/pkg/crypto/x509/#CertPool.AppendCertsFromPEM

It should be either checked or explicitly discarded.

File: util.go (hunk 1 above)
Comment (ridv) [reply util.go]: Good catch, wrapped the error now.

File: util.go (hunk 2 above)
Comment (ridv) [reply util.go]: Using it to detect cases where no certificates get loaded into the pool and returning an error based upon that.

**Pull request merged**